live-build/misc/migration-scripts/dx_execute

```diff
@@ -16,9 +16,71 @@
 #
 
 #
-# [WIP]
-# The existence of this file is checked by dx_unpack, thus we provide
-# it as an empty file until the needed functionality is added.
+# Sets up the FreeBSD bootloader to boot into that Linux image next time
+# we reboot.
 #
 
+set -o pipefail
+
+function die() {
+	report "$(basename "$0"): $*" >&2
+	exit 1
+}
+
+[[ "$(uname -s)" == "SunOS" ]] || die "script can only be used in illumos"
+
+#
+# Undo any existing default timeout commands in the bootloader's menu.
+#
+sed -i '/menu_timeout_command/d' /boot/menu.rc.local
+
+#
+# Get the RDS and the current ZFS pool that we're currently using.
+#
+RDS=$(mount | awk '/^\/ /{ print $3 }')
+RPOOL=${RDS%%/*}
+
+#
+# Check that the version to which we are upgrading is compatible with the
+# currently installed version. (Look at version.info to see if we are
+# above the minimum version required. Since we are doing a migration the
+# minimum version should be the latest 5.3.
+#
+# CURRENT_DDS=$(dirname "$(mount | awk '/^\/opt\/delphix /{ print $3 }')")
+# CURRENT_VERSION=$(basename $current_dds)
+#
+# WIP: check if current version is 5.3, if not bail
+#
+
+#
+# Ensure that the expected Linux dataset layout exists.
+#
+[[ $(zfs list -o name -Hr rpool/ROOT | wc -l) -eq 2 ]] ||
```

Comment:
`rpool/ROOT` -> `"$RPOOL/ROOT"`
We should stay consistent. I'm wondering if we shouldn't just use `rpool` everywhere. I can't think of situations where the root pool is not called `rpool`.